lib/models/room.js

```diff
@@ -401,6 +410,27 @@ Room.prototype.addReceipt = function(event) {
     });
 };
 
+/**
+ * Update the room-tag event for the room.  The previous one is overwritten.
+ * @param {MatrixEvent} event the m.tag event
+ */
+Room.prototype.addTags = function(event) {
+    // event content looks like:
+    // content: {
+    //    tags: {
+    //       $tagname: { $metadata: $value },
+    //       $tagname: { $metadata: $value },
+    //    }
+    // }
+
+    // XXX: we could do a deep-comparison to see if the tags have really
+    // changed - but do we want to bother?
+    this.emit("Room.tags", this);
+
+    // XXX: do we need to deep copy here?
+    this._tags = event.content.tags;
```

Comment:
This isn't true of anything else in the Room class, as far as I can see - I implemented it like this originally and then changed it to match the rest of the class.  Unsure if I want to break symmetry for this.

the getContent() thing is fixed.